pkg/aws/client.go

```diff
@@ -441,42 +443,45 @@ func (c *awsClient) FetchPublicSubnetMap(subnets []*ec2.Subnet) (map[string]bool
 	if len(subnets) == 0 {
 		return mapSubnetIdToPublic, nil
 	}
-	subnetIds := []*string{}
-	for _, subnet := range subnets {
-		subnetIds = append(subnetIds, subnet.SubnetId)
-	}
-	routeTablesResp, err := c.ec2Client.DescribeRouteTables(&ec2.DescribeRouteTablesInput{
-		Filters: []*ec2.Filter{
-			{
-				Name:   aws.String("association.subnet-id"),
-				Values: subnetIds,
-			},
-		},
-	})
-	if err != nil {
-		return mapSubnetIdToPublic, err
-	}
+	// AWS has a limit of 200 filters per query so it needs to be broken up in chunks
+	chunks := helper.ChunkSlice(subnets, awsMaxFilterLength)
 
-	if routeTablesResp == nil {
-		return mapSubnetIdToPublic, fmt.Errorf(
-			"No route table found for associated subnets '%s'",
-			helper.SliceToSortedString(aws.StringValueSlice(subnetIds)),
-		)
-	}
-
-	for _, routes := range routeTablesResp.RouteTables {
-		for _, association := range routes.Associations {
-			subnetAssociation := aws.StringValue(association.SubnetId)
-			mapSubnetIdToPublic[subnetAssociation] = false
-			for _, route := range routes.Routes {
-				if strings.HasPrefix(aws.StringValue(route.GatewayId), "igw") {
-					// There is no direct way in the AWS API to determine if a subnet is public or private.
-					// A public subnet is one which has an internet gateway route
-					// we look for the gatewayId and make sure it has the prefix of igw to differentiate
-					// from the default in-subnet route which is called "local"
-					// or other virtual gateway (starting with vgv)
-					// or vpc peering connections (starting with pcx).
-					mapSubnetIdToPublic[subnetAssociation] = true
+	for _, curChunk := range chunks {
+		subnetIds := []*string{}
+		for _, subnet := range curChunk {
+			subnetIds = append(subnetIds, subnet.SubnetId)
+		}
+		routeTablesResp, err := c.ec2Client.DescribeRouteTables(&ec2.DescribeRouteTablesInput{
+			Filters: []*ec2.Filter{
+				{
+					Name:   aws.String("association.subnet-id"),
+					Values: subnetIds,
+				},
+			},
+		})
+		if err != nil {
+			return mapSubnetIdToPublic, err
+		}
+		if routeTablesResp == nil {
```

Comment:
Should we continue to the next chunk in this case?

Reply:
I've thought about it, I'm not so sure. If we skip it might be confusing for the customer too, I'd rather we merge as is and check on user behavior :thinking:

Reply:
I think if we can help the user with a slightly more informative error message that would be good. I kinda feel like most customers are going to react with "And what do I do now?" when they see a message about there being no route table associated with a subnet. 

Is this a problem they can fix with ROSA CLI or something they need to talk to their AWS account owner about for example.

Reply:
I checked and it seems like " Each subnet in your VPC must be associated with a route table, which controls the routing for the subnet (subnet route table)" so I think it's ok to return an error here after a second thought. https://docs.aws.amazon.com/vpc/latest/userguide/VPC_Route_Tables.html